### PR TITLE
refactor(AWS): Update AWS Amazon MQ module

### PR DESCRIPTION
**Pull request by YKharouni**
- Refactor AWS Amazon MQ module
- add simple and complete deployment examples of Amazon MQ (RabbitMQ and Amazon MQ)

### CODE REVIEW

File: storage/aws/mq/main.tf [hunk 1]
```diff
@@ -1,25 +1,56 @@
+locals {
+  tags = merge(var.tags, { module = "amazon-mq" })
+  subnet_ids = (var.deployment_mode == "SINGLE_INSTANCE" ? [var.vpc_subnet_ids[0]] : [
+    var.vpc_subnet_ids[0],
+    var.vpc_subnet_ids[1],
+  ])
+  username = can(coalesce(var.username)) ? var.username : random_string.user.result
+  password = can(coalesce(var.password)) ? var.password : random_password.password.result
```
Comment (lemaitre-aneo) [inline storage/aws/mq/main.tf:8]:
```suggestion
  username = try(coalesce(var.username), random_string.user.result)
  password = try(coalesce(var.password), random_password.password.result)
```

File: storage/aws/mq/variables.tf [hunk 2]
```diff
@@ -13,40 +13,98 @@ variable "name" {
...
+}
+
+variable "vpc_cidr_blocks" {
+  description = "AWS VPC cidr block"
+  type        = list(string)
+}
+
+variable "vpc_subnet_ids" {
+  description = "AWS VPC subnet ids"
+  type        = list(string)
```
Comment (lemaitre-aneo) [inline storage/aws/mq/variables.tf:28]: Same remark as for EFS

File: storage/aws/mq/versions.tf [hunk 3]
```diff
@@ -3,11 +3,11 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 5.3.0"
+      version = ">= 5.4.0"
     }
     random = {
       source  = "hashicorp/random"
-      version = ">= 3.4.3"
+      version = "~> 3.5.1"
```
Comment (lemaitre-aneo) [inline storage/aws/mq/versions.tf:10]: The recommended way is to let precise version binding decision to the root module.

```suggestion
      version = "=> 3.5.1"
```

File: storage/aws/mq/main.tf (hunk 1 above)
Comment (YKharouni) [reply storage/aws/mq/main.tf:8]: done !

File: storage/aws/mq/versions.tf (hunk 3 above)
Comment (YKharouni) [reply storage/aws/mq/versions.tf]: done

**Review by alouvion-aneo — APPROVED** [review]

**Pull request merged**